crypten/cryptensor.py

```diff
@@ -231,6 +235,21 @@ def detach(self):
         clone.requires_grad = False
         return clone
 
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        """Allows torch static functions to work on CrypTensors."""
+        if kwargs is None:
+            kwargs = {}
+        func_names = {getattr(torch, name): name for name in STATIC_FUNCTIONS}
```

Comment:
Q: Is ```func_names``` constructed every time we call ```cat```/```stack```?
If that is the case maybe we can make this to be made only once.

Reply:
It is constructed every time the user calls `torch.some_function(cryptensor_inputs)` (although only `torch.stack` and `torch.cat` work at the moment). I can make this a global variable just like `STATIC_FUNCTIONS`.

Reply:
Maybe it is worth making it global in case more functions are to be added? (depending on the CrypTen RoadMap)

An idea might be to have something like:

```
TORCH_TO_CRYPTEN_FUNCS = { # Might not be the best name
        torch.stack: crypten.stack, 
        torch.cat: crypten.cat, 
    }
```
^^ This will require for the ```crypten module``` to be loaded when the ```dict``` is created (this might create a cyclic dependency?).

Another idea is to create a ```dict``` like it is already implemented in this PR.
```Dict[torch_func, func_name]``` and do the ```getattr``` using the function name

Reply:
This is more or less what I did in https://github.com/lvdmaaten/CrypTen/commit/a20a21d8553d04434569b2f0e6d06c9975c87189 although I map from PyTorch function to CrypTen function name to circumvent the circular dependency you describe.